include/mqtt/client.hpp

```diff
@@ -1378,14 +1388,94 @@ class client : public endpoint<std::mutex, std::lock_guard, PacketIdBytes> {
         set_timer();
     }
 
+    static optional<std::uint32_t> get_session_expiry_interval_by_props(v5::properties const& props) {
```

Comment:
I know that the standard explicitly says it's a 32 bit number, but I think we should make this a typedef. E.g "SessionExpiryInterval_t"

Reply:
Implemented.